src/algebra/ordered_ring.lean

```diff
@@ -870,6 +870,14 @@ end
 lemma abs_le_one_iff_mul_self_le_one : abs a ≤ 1 ↔ a * a ≤ 1 :=
 by simpa only [abs_one, one_mul] using @abs_le_iff_mul_self_le α _ a 1
 
+@[simp]
+lemma sub_le_sub_flip : a - b ≤ b - a ↔ a ≤ b :=
```

Comment:
Does anyone know a proof of this without the `ring` structure?

Reply:
This works:
```lean
import algebra.ordered_group

variables (α : Type*) [linear_ordered_add_comm_group α] {a b c d : α}

@[simp]
lemma sub_le_sub_flip : a - b ≤ b - a ↔ a ≤ b :=
begin
  rw [sub_le_iff_le_add, sub_add_eq_add_sub, le_sub_iff_add_le],
  split,
  { intro h,
    by_contra H,
    rw not_le at H,
    apply not_lt.2 h,
    exact add_lt_add H H, },
  { intro h,
    exact add_le_add h h, }
end
```

Reply:
Thanks!